### PR TITLE
Add support for CompressedImage and CameraInfo

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

File: depthai_ros_driver/include/depthai_ros_driver/conversion.hpp [hunk 1]
```diff
@@ -99,16 +114,26 @@ struct adapt_dai2ros<depthai_datatype_msgs::RawImgFrame> {
         bridge.header.stamp.sec = input.ts.sec;
         bridge.header.stamp.nsec = input.ts.nsec;
 
-        pub.image_pub.publish(bridge.toImageMsg());
-        sensor_msgs::CameraInfo info;  // @TODO(kunaltyagi): get camera info from somewhere
-        pub.camera_info_pub.publish(info);
+        // @TODO(kunaltyagi): switch between raw and compressed publishers
```
Comment (kunaltyagi) [inline depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: @ardiya How can I get this information?

**Comment by kunaltyagi** [comment]
Support for default camera info to be next, but this should allow basic work to be done

File: depthai_ros_driver/include/depthai_ros_driver/conversion.hpp (hunk 1 above)
Comment (ardiya) [reply depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: `template <class MsgType, bool is_t_compressed=false>` in adapt_dai2ros and in generate_pub_lambda?

Then call generate_pub_lambda when it's node is coming from dai::node::VideoEncoder? 
```
        case dai::DatatypeEnum::ImgFrame:
                if (node_is_coming_from_encoder) {
                  generate_pub_lambda<depthai_datatype_msgs::RawImgDetections, true>(...)
                } else {
                  generate_pub_lambda<depthai_datatype_msgs::RawImgDetections, false>(...)
                }
```
It feels like you told me this last time, no?

Comment (kunaltyagi) [reply depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: Can this be inferred at runtime as well?

Comment (ardiya) [reply depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: if you use std::any, it's possible to have both image and compressedimage at runtime?
Might not need extra template parameter if we use std::any.
I never really use it though

Comment (kunaltyagi) [reply depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: I was wondering if the output we read can be used to identify if it's compressed

Comment (ardiya) [reply depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: Oh, the compressed image gonna be `dai::RawImgFrame::Type::BITSTREAM` type.
https://github.com/rapyuta-robotics/depthai_ros/blob/gen2_ros/depthai_ros_driver/src/conversion.cpp#L94-L98
I haven't handle this as special case yet.

File: depthai_ros_driver/include/depthai_ros_driver/conversion.hpp [hunk 2]
```diff
@@ -79,36 +96,54 @@ struct adapt_dai2ros<depthai_datatype_msgs::RawImgFrame> {
...
-                nh.advertise<OutputType>(name + "/image_raw", q_size);  // @TODO(kunaltyagi): compressed if needed
+        pubs.raw_image_pub = nh.advertise<OutputType>(name + "/image_raw", q_size);
+        pubs.compressed_image_pub = nh.advertise<OutputType>(name + "/image_raw/compressed", q_size);
         pubs.camera_info_pub = nh.advertise<sensor_msgs::CameraInfo>(name + "/camera_info", q_size);
+
+        std::string camera_name = "default";
+        if (!nh.getParam("camera_name", camera_name)) {
+            nh.setParam("camera_name", camera_name);
+        }
+        const auto uri = camera_param_uri + camera_name + "/" + name + ".yaml";
```
Comment (ardiya) [inline depthai_ros_driver/include/depthai_ros_driver/conversion.hpp:144]:
```suggestion
        const auto uri = camera_param_uri + "/" + camera_name + "/" + name + ".yaml";
```
I noticed this when trying to convert depth+camera_info to depth

Comment (ardiya) [inline depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]:
```suggestion
        const auto uri = camera_param_uri + "/" + camera_name + "/" + name + ".yaml";
```

File: depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
 #include <ros/publisher.h>
 #include <ros/ros.h>
-#include <ros/callback_queue.h>
 
-#include <range/v3/view.hpp>
+// lib headers
 #include <range/v3/algorithm/find.hpp>
+#include <range/v3/view.hpp>
+
+// std headers
```
Comment (ardiya) [inline depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:21]: Do we have any convention for order of the include?

Google, [for example](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) include C header, C++ header, 3rd party header, followed by project header.

**Review by ardiya — APPROVED** [review]
LGTM

File: depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp (hunk 3 above)
Comment (kunaltyagi) [reply depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:21]: I use increasing order of trust of no error:
* project headers
* other library headers
* famous library headers
* cpp headers
* c headers

This ensures that bad includes are detected asap. Ofc, if the includes have some constraint on the order due to macros, nothing can be done

File: depthai_ros_driver/include/depthai_ros_driver/conversion.hpp [hunk 4]
```diff
@@ -1,28 +1,45 @@
...
+    std::shared_ptr<camera_info_manager::CameraInfoManager> info_manager_ptr;
 };
 
 auto getNumSubscribers(const ros::Publisher& pub) {
     return pub.getNumSubscribers();
 }
 
 auto getNumSubscribers(const ImagePublishers& pubs) {
-    return pubs.image_pub.getNumSubscribers() + pubs.camera_info_pub.getNumSubscribers();
+    return pubs.raw_image_pub.getNumSubscribers() + pubs.camera_info_pub.getNumSubscribers();
```
Comment (kota-rr) [inline depthai_ros_driver/include/depthai_ros_driver/conversion.hpp]: + compressed_image_pub

**Review by kota-rr — APPROVED** [review]

**Pull request merged**